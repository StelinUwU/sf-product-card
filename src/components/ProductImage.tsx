import React, { useContext } from 'react';
import { ProductContext } from './ProductCard';
import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';

export interface Props {
  className?: string;
  img?: string;
  style?: React.CSSProperties;
}

export const ProductImage = ({ img, className, style }: Props) => {
  const { product } = useContext(ProductContext);
  const image = img || product.img || noImage;
  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={image}
      style={style}
      alt="Product img"
    />
  );
};
