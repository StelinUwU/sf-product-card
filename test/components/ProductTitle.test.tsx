import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product } from '../data/products';

describe('ProductTitle', () => {
  test('should render the component according to the props', () => {
    const wrapper = renderer.create(
      <ProductTitle title="CustomProduct" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render the component with the product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
