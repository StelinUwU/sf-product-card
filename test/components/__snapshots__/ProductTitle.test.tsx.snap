// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should render the component according to the props 1`] = `
<span
  className="productDescription custom-class"
>
  CustomProduct
</span>
`;

exports[`ProductTitle should render the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug without image
  </span>
</div>
`;
